app/app/Board.js

```diff
@@ -187,4 +187,60 @@ export class Board {
 
         this.deleteControls(controlsToDelete);
     }
-}
+
+    /**
+     * Initiate Selection Area drawing process.
+     * @param x
+     * @param y
+     */
+    startDrawingSelectionArea(x, y) {
+        this.isDrawingSelectionArea = true;
+
+        this.startXSelectionArea = x;
+        this.startYSelectionArea = y;
+
+        this.selectionArea = this.paper.rect(x, y, 0, 0);
+        this.selectionArea.remove();
```

Comment:
If I understood correctly, the rect object is always being removed and recreated again on every mousemove event. Right? How about just changing the properties and redrawing it?
For connection line I did something similar. Could you check it?